src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -388,7 +389,9 @@ Result SpatialJoinAlgorithms::LibspatialjoinAlgorithm() {
     for (size_t i = 0; i < results[t].size(); i++) {
       const auto& res = results[t][i];
       double dist = 0;
-      if (withinDist >= 0) dist = resultDists[t][i];
+      if (joinTypeVal == SpatialJoinType::WITHIN_DIST) {
+        dist = resultDists[t][i];
+      }
```

Comment:
What about the `withinDist = -1` above? Since you still need it (because `libspatialjoin` needs it), please add this information to the comment before the `withinDist = -1` assignment